### PR TITLE
Refactor environment

### PR DESCRIPTION
**Pull request by florian-huber**
- remove ``gensim`` dependency
- add support for ``Python 3.8`` (including CI tests)
- lower allowed number of "uncovered members" to 20

(see also #144 )

### CODE REVIEW

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Developer install and tests work with py3.8
Conda build works with py3.7 conda env without upgrading to py38, as expected

Can you add 3.8 to setup.py, so supported versions are mentioned everywhere?

To reduce number of CI jobs run python versions only one ubuntu and run 3.8 on Windows and OSX.

File: readthedocs/index.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ Installation
 
 Prerequisites:  
 
-- Python 3.7  
+- Python 3.7 or 3.8
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply readthedocs/index.rst:31]: Good point, done.

**Comment by florian-huber** [comment]
Many thanks @sverhoeven . 
I did add the change to setup.py.
And I also changed my push vs pull_request solution to the one you suggested by mail. (I had missed that any merged PR will anyway trigger a ``push`` so we do not need to take care of that)

[edit]
I did partly follow your advice to further reduce the number of CI runs. We now have
- 4 analyze runs (Python 3.7 + ubuntu and Python 3.8 for ubuntu, mac-os, and windows)
- 3 build runs (Python 3.7)
- 6 verify runs (Python 3.7 and 3.8 for ubuntu, mac-os, and windows)

**Review by sverhoeven — APPROVED** [review]

**Comment by florian-huber** [comment]
Thanks again Stefan for the reviewing!

**Pull request merged**